src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/AlwaysGenerateDocumentsService.java

```diff
@@ -41,25 +42,20 @@ public AlwaysGenerateDocumentsService(
     }
 
     @Override
-    public byte[] generateClaimIssueReceipt(String externalId, String authorisation) {
-        return claimIssueReceiptService.createPdf(getClaimByExternalId(externalId, authorisation));
+    public byte[] generateDocument(String externalId, ClaimDocumentType claimDocumentType, String authorisation) {
+        switch (claimDocumentType) {
+            case CLAIM_ISSUE_RECEIPT:
+                return claimIssueReceiptService.createPdf(getClaimByExternalId(externalId, authorisation));
+            case SEALED_CLAIM:
+                return sealedClaimPdfService.createPdf(getClaimByExternalId(externalId, authorisation));
+            case DEFENDANT_RESPONSE_RECEIPT:
+                return defendantResponseReceiptService.createPdf(getClaimByExternalId(externalId, authorisation));
+            case SETTLEMENT_AGREEMENT:
+                return settlementAgreementCopyService.createPdf(getClaimByExternalId(externalId, authorisation));
+        }
+        throw new RuntimeException(); //todo proper exception
```

Comment:
How about adding a default case a throw IllegalArgumentException for `claimDocumentType`?

Reply:
yup